build.ps1

```diff
@@ -1,12 +1,20 @@
+$SCRIPT_ROOT = "$PSScriptRoot"
+$SdkVersion = Get-Content (Join-Path $SCRIPT_ROOT "DotnetCLIVersion.txt")
+
 $env:DOTNET_CLI_TELEMETRY_OPTOUT = 1
 $env:DOTNET_SKIP_FIRST_TIME_EXPERIENCE = 1
+$env:NUGET_PACKAGES = "$ScriptRoot\packages\"
+
+& "$SCRIPT_ROOT\init-tools.cmd"
+
+$CLIPATH = "$SCRIPT_ROOT\Tools\dotnetcli"
+$SDKPATH = "$CLIPATH\sdk\$SdkVersion"
 
-$ScriptRoot = "$PSScriptRoot"
-$SdkVersion = Get-Content (Join-Path $ScriptRoot ".cliversion")
+& "$CLIPATH\dotnet" restore "$SCRIPT_ROOT\tasks\Microsoft.DotNet.SourceBuild.Tasks\Microsoft.DotNet.SourceBuild.Tasks.csproj"
+& "$CLIPATH\dotnet" build "$SCRIPT_ROOT\tasks\Microsoft.DotNet.SourceBuild.Tasks\Microsoft.DotNet.SourceBuild.Tasks.csproj"
 
-& "$ScriptRoot\bootstrap\bootstrap.ps1"
-& "$ScriptRoot\Tools\dotnetcli\dotnet" restore "$ScriptRoot\tasks\Microsoft.DotNet.SourceBuild.Tasks\Microsoft.DotNet.SourceBuild.Tasks.csproj"
-& "$ScriptRoot\Tools\dotnetcli\dotnet" build "$ScriptRoot\tasks\Microsoft.DotNet.SourceBuild.Tasks\Microsoft.DotNet.SourceBuild.Tasks.csproj"
-& "$ScriptRoot\Tools\dotnetcli\dotnet" "$ScriptRoot\Tools\dotnetcli\sdk\$SdkVersion\MSBuild.dll" "$ScriptRoot\build.proj" @args /t:WriteDynamicPropsToStaticPropsFiles /p:GeneratingStaticPropertiesFile=true
-& "$ScriptRoot\Tools\dotnetcli\dotnet" "$ScriptRoot\Tools\dotnetcli\sdk\$SdkVersion\MSBuild.dll" "$ScriptRoot\build.proj" @args
+Remove-Item -Recurse -Force "$NUGET_PACKAGES"
```

Comment:
Why do we need to remove the packages here?

Reply:
The quick answer: Because we do in `build.sh`.

The longer answer relates to our discussion today.  Restoring BuildTools will pollute the NUGET_PACKAGES cache by installing packages that we are going to build ourselves during the build.  And then later, when downstream repos want to consume the package, they will get the one that was restored by BuildTools (which isn't what we built).

Reply:
OK I'm fine with consistency between build.cmd/sh for now but we should isolate these and not have to clean these like this.